### PR TITLE
INSPIR-192 Add (non-)core reference counts to the workflow

### PR DESCRIPTION
**Pull request by michamos**
## Description

Adds the reference counts to the workflow and refactors some useful utils and related tests.

## Checklist:

- [ ] I have all the information that I need (if not, move to `RFC` and look for it).
- [ ] I linked the related issue(s) in the corresponding commit logs.
- [ ] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [ ] My code follows the code style of this project.
- [ ] I've added any new docs if API/utils methods were added.
- [ ] I have updated the existing documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: inspirehep/modules/records/serializers/schemas/json.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def processed_refs(self, data):
             data['metadata']['references'] = []
             return data
 
-        reference_records = self.resolve_records(references)
+        reference_records = self.resolve_records(data)
```
Comment (drjova) [inline inspirehep/modules/records/serializers/schemas/json.py]: hmm should be ``data['metadata']``, no? cause you have [this](https://github.com/inspirehep/inspire-next/pull/3509/files#diff-65ba6e9a03cccd086772d35d5fd5ba60R119):

```python
resolved_records = get_linked_records_in_field(data, 'references.record')
```

Comment (michamos) [reply inspirehep/modules/records/serializers/schemas/json.py]: it passes all tests though, so I guess this is not really tested.

Comment (michamos) [reply inspirehep/modules/records/serializers/schemas/json.py]: I changed the unit test mocking the `get_linked_records_in_field` function into an integration test without mocks. With @turtle321's `TestRecordMetadata` factory, it's almost easier to create test data in the db than mocking functions reading from the db.

**Comment by michamos** [comment]
ci merge please

**Comment by michamos** [comment]
ci merge please

**Pull request merged**